internal/service/federatedsettingsidentityprovider/resource_federated_settings_identity_provider.go

```diff
@@ -141,13 +141,9 @@ func resourceMongoDBAtlasFederatedSettingsIdentityProviderRead(ctx context.Conte
 		return diag.FromErr(fmt.Errorf("error setting sso url (%s): %s", d.Id(), err))
 	}
 
-	if err := d.Set("id", federatedSettingsIdentityProvider.Id); err != nil {
-		return diag.FromErr(fmt.Errorf("error setting Id (%s): %s", d.Id(), err))
-	}
-
 	d.SetId(conversion.EncodeStateID(map[string]string{
 		"federation_settings_id": federationSettingsID,
-		"okta_idp_id":            oktaIdpID,
+		"okta_idp_id":            federatedSettingsIdentityProvider.Id,
```

Comment:
This is in the flow where the user is already using the new ID, so now uses the new ID in the TF state ID

Reply:
How do you undersatnd that "this is in the flow"?